ostia-operator/pkg/apicast/reconcile.go

```diff
@@ -0,0 +1,93 @@
+package apicast
+
+import (
+	"github.com/3scale/ostia/ostia-operator/pkg/apis/ostia/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk/action"
+	"github.com/operator-framework/operator-sdk/pkg/sdk/query"
+	log "github.com/sirupsen/logrus"
+	"reflect"
+)
+
+//Reconcile takes care of the main apicast reconciliation loop
+func Reconcile(api *v1alpha1.API) (err error) {
+
+	log.Infof("[%s] Got API Object: %s", api.Namespace, api.Name)
+
+	// Reconcile DeploymentConfig object
+	err = reconcileDeploymentConfig(api)
+	if err != nil {
+		log.Errorf("[%s] API: %s Failed to reconcile DeploymenConfig %v", api.Namespace, api.Name, err)
+	}
+
+	// Reconcile Service object
+	err = reconcileService(api)
+	if err != nil {
+		log.Errorf("[%s] API: %s Failed to reconcile Service %v", api.Namespace, api.Name, err)
+	}
```

Comment:
is it ok to keep calling reconcile* calls if the first one or calls in the middle fail?